core/services/relay/evm/bindings.go

```diff
@@ -5,57 +5,63 @@ import (
 	"fmt"
 
 	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
 )
 
-// key is contract name
-type contractBindings map[string]readBindings
+// bindings manage all contract bindings, key is contract name.
+type bindings map[string]*contractBinding
```

Comment:
this is unsafe across go routines. do you need locks to prevent races between the access methods? if not, why not (worth an explicit comment)?

Reply:
As long as we aren't replacing structs under map keys this shouldn't be. dangerous to read. If by access methods you mean GetReadBinding it doesn't need locks since it can't be called before AddReadBinding as they are added on startup

Reply:
Might be worth a comment somewhere warning that it's not safe to call `AddReadBindings()` after initial startup. (And maybe also, that it's not safe to call `GetReadBindings()` before startup is complete.) If someone ever decides to add support for dynamically updating the bindings, that would help them realize they need to add a mutex